### PR TITLE
ENH+BF: Provide <fail2ban-agent-string> tag

### LINKED ISSUE #1271: In action "blocklist_de.conf" is hardcoded user-agent "fail2ban v0.8.12"
Version 0.9.2 and current dev code

In file
/etc/fail2ban/action.d/blocklist_de.conf
is the line
```
actionban = curl --fail --data-urlencode 'server=<email>' --data 'apikey=<apikey>' --data 'service=<service>' --data 'ip=<ip>' --data-urlencode 'logs=<matches>' --data 'format=text' --user-agent "fail2ban v0.8.12" "https://www.blocklist.de/en/httpreports.html"
```
The user-agent setting is wrong.
I would suggest to determine the fail2ban version on runtime and use that value instead of hardcoded version, since, like in this case, it will probably be wrong in the future.

### PR DESCRIPTION
**Pull request by yarikoptic**
Was crafted to resolve #1271 but actually not sure now if is worth this approach over "let's execute fail2ban-client version" and define that variable for string interpolations in e.g. actions/common.conf

### CODE REVIEW

File: fail2ban/version.py [hunk 1]
```diff
@@ -25,3 +25,5 @@
 __license__ = "GPL-v2+"
 
 version = "0.9.3.dev"
+
+agent_string = "Fail2Ban %s" % version
```
Comment (ranvis) [inline fail2ban/version.py]: Perhaps we can use "Fail2Ban/%s" to confirm [RFC 7231 5.5.3](https://tools.ietf.org/html/rfc7231#section-5.5.3)?

Comment (yarikoptic) [inline fail2ban/version.py]: On Fri, 18 Dec 2015, SATŌ Kentarō wrote:

> In [1]fail2ban/version.py:

> > @@ -25,3 +25,5 @@
> >  __license__ = "GPL-v2+"

> >  version = "0.9.3.dev"
> > +
> > +agent_string = "Fail2Ban %s" % version

> Perhaps we can use "Fail2Ban/%s" to confirm [2]RFC 7231 5.5.3?
Great idea ;)   pushed update

**Comment by yarikoptic** [comment]
@fail2ban/developers , @fail2ban/contributors -- any feedback?

**Comment by sebres** [comment]
I don't like it - too many changes (and conflicts :)) for one little thing (a new parameter used in action), that can be also easy added somewhere in configreader (ActionReader or JailReader)...
Will try to check it now...

**Comment by yarikoptic** [comment]
I don't quite like it as well, but not for "too many changes" of which I see not that many ;-)  Will wait then for your version

**Comment by sebres** [comment]
You are welcome: https://github.com/fail2ban/fail2ban/compare/master...sebres:bf-fail2ban-version

The whole changes are in  `fail2ban/client/jailreader.py` (and its usage in configs).
But because we had at all no test cases for `action_blocklist_de` and `action_badips` the test case grows a little bit...

Should I make a PR?

**Comment by sebres** [comment]
Ah yeah, by the way, badips definition in jail.conf had an interpolation bug, because `%(name)s` not available:
```dif
-action_badips = badips.py[category="%(name)s", banaction="%(banaction)s"]
+action_badips = badips.py[category="%(__name__)s", banaction="%(banaction)s", agent="%(fail2ban_agent)s"]
```

**Comment by yarikoptic** [comment]
PR would be much better for testing and review. Cheers!

**Pull request closed without merging**